File: frontend/lib/models/user_model.dart
class UserModel {
  final String collegeId;
  final String name;
  final String gender;
  final String? phone;
  final String? fcmToken;
  final DateTime createdAt;

  UserModel({
    required this.collegeId,
    required this.name,
    required this.gender,
    this.phone,
    this.fcmToken,
    required this.createdAt,
  });

  factory UserModel.fromJson(Map<String, dynamic> json) {
    return UserModel(
      collegeId: json['college_id'],
      name: json['name'] ?? json['college_id'],
      gender: json['gender'],
      phone: json['phone'],
      fcmToken: json['fcm_token'],
      createdAt: DateTime.parse(json['created_at']),
    );
  }

  Map<String, dynamic> toJson() => {
    'college_id': collegeId,
    'name': name,
    'gender': gender,
    'phone': phone,
    'fcm_token': fcmToken,
    'created_at': createdAt.toIso8601String(),
  };

  UserModel copyWith({
    String? name,
    String? phone,
    String? fcmToken,
  }) {
    return UserModel(
      collegeId: collegeId,
      name: name ?? this.name,
      gender: gender,
      phone: phone ?? this.phone,
      fcmToken: fcmToken ?? this.fcmToken,
      createdAt: createdAt,
    );
  }
}

File: backend/app/models/user_model.py
# backend/models/user_model.py

from datetime import datetime
from typing import Optional
from pydantic import BaseModel

class UserModel(BaseModel):
    college_id: str
    name: str
    gender: str  # 'male', 'female', 'other'
    phone: Optional[str] = None
    fcm_token: Optional[str] = None
    created_at: datetime

    @classmethod
    def from_dict(cls, data: dict):
        return cls(
            college_id=data.get('college_id', ''),
            name=data.get('name', ''),
            gender=data.get('gender', ''),
            phone=data.get('phone'),
            fcm_token=data.get('fcm_token'),
            created_at=datetime.fromisoformat(data.get('created_at', datetime.now().isoformat()))
        )
    
    def to_dict(self) -> dict:
        return {
            'college_id': self.college_id,
            'name': self.name,
            'gender': self.gender,
            'phone': self.phone,
            'fcm_token': self.fcm_token,
            'created_at': self.created_at.isoformat()
        }